lockbox-ios/View/RootView.swift

```diff
@@ -66,66 +111,93 @@ class RootView: UIViewController, RootViewProtocol {
         return self.currentViewController?.presentedViewController is UINavigationController
     }
 
-    func startMainStack<T: UINavigationController>(_ type: T.Type) {
-        self.currentViewController = type.init()
+    func startMainStack<T: UIViewController>(_ viewController: T) {
+        self.currentViewController = viewController
     }
 
     func startModalStack<T: UINavigationController>(_ navigationController: T) {
-        self.currentViewController?.present(navigationController, animated: true)
+        self.present(navigationController, animated: !isRunningTest, completion: nil)
     }
 
     func dismissModals() {
-        self.currentViewController?.presentedViewController?.dismiss(animated: !isRunningTest, completion: nil)
+        self.presentedViewController?.dismiss(animated: !isRunningTest, completion: nil)
     }
 
     func pushLoginView(view: LoginRouteAction) {
         switch view {
         case .welcome:
-            self.currentViewController?.popToRootViewController(animated: !isRunningTest)
+            self.currentNaivgationController?.popToRootViewController(animated: !isRunningTest)
         case .fxa:
-            self.currentViewController?.pushViewController(FxAView(), animated: !isRunningTest)
+            self.currentNaivgationController?.pushViewController(FxAView(), animated: !isRunningTest)
         case .onboardingConfirmation:
             if let onboardingConfirmationView = UIStoryboard(name: "OnboardingConfirmation", bundle: nil).instantiateViewController(withIdentifier: "onboardingconfirmation") as? OnboardingConfirmationView {
-                self.currentViewController?.pushViewController(onboardingConfirmationView, animated: !isRunningTest)
+                self.currentNaivgationController?.pushViewController(onboardingConfirmationView, animated: !isRunningTest)
             }
         case .autofillOnboarding:
             if let autofillOnboardingView = UIStoryboard(name: "AutofillOnboarding", bundle: nil).instantiateViewController(withIdentifier: "autofillonboarding") as? AutofillOnboardingView {
-                self.currentViewController?.pushViewController(autofillOnboardingView, animated: !isRunningTest)
+                self.currentNaivgationController?.pushViewController(autofillOnboardingView, animated: !isRunningTest)
             }
         case .autofillInstructions:
             if let autofillInstructionsView = UIStoryboard(name: "SetupAutofill", bundle: nil).instantiateViewController(withIdentifier: "autofillinstructions") as? AutofillInstructionsView {
-                self.currentViewController?.pushViewController(autofillInstructionsView, animated: !isRunningTest)
+                self.currentNaivgationController?.pushViewController(autofillInstructionsView, animated: !isRunningTest)
+            }
+        }
+    }
+
+    func pushSidebarView(view: MainRouteAction) {
```

Comment:
These `pushXView` methods that take a `RouteAction` are getting more & more unwieldy... I wonder if it would make more sense to move away from that paradigm and have the `RoutePresenter` do the view instantiation itself as you're doing already with the `startMainStack` etc methods...

Reply:
So `RootView` would expose `pushView(vc: ViewController)`, `pushSidebarView(vc: ViewController)`, `popView()` and `popToRoot()` ?

Reply:
`pushSidebarView` can probably be removed for now since we don't actually push anything there.

Reply:
Yeah, that's what I was thinking. What do you think about that? Definitely open to opinions here but I think it would clean up a lot of the `RootView` code without introducing too much complexity in the `RootPresenter`.

Reply:
@sashei I updated the PR - do you mind taking a look at the new format? ViewControllers are built using a ViewFactory in `RootPresenter` and then passed to `RootView` for display.